File: cukcuk clone/app/src/main/java/vn/com/misa/cukcukstarterclone/data/source/IUserDataSource.java
package vn.com.misa.cukcukstarterclone.data.source;

import android.app.Activity;

import vn.com.misa.cukcukstarterclone.data.IOnLoadedCallback;

/**
 * @created_by KhanhNQ on 08-Feb-2021.
 */
public interface IUserDataSource {
    interface Local {

    }

    interface Remote {
        void login(String username, String password, IOnLoadedCallback<Boolean> callback);

        void register( String username, String password, IOnLoadedCallback<Boolean> callback);
    }
}

File: app/src/main/java/com/misa/fresher/data/source/user/IUserDataSource.kt
package com.misa.fresher.data.source.user

import com.misa.fresher.data.IOnLoadedCallback
import com.misa.fresher.data.model.User
import com.misa.fresher.data.model.UserError

/**
 * - interface's purpose:
 *
 * @author HTLong
 * @edit_at 2022-03-21
 */

interface IUserDataSource {
    class Local {

    }

    interface Remote {
        fun signIn(username: String, password: String, callback: IOnLoadedCallback<User, UserError>)
        fun signUp(username: String, password: String, callback: IOnLoadedCallback<User, UserError>)
    }
}